src/pull_request.py

```diff
@@ -0,0 +1,237 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module for handling git repository."""
+
+import logging
+import re
+from pathlib import Path
+
+from git import GitCommandError
+from git.repo import Repo
+from github import Github
+from github.GithubException import GithubException
+from github.Repository import Repository
+
+from .exceptions import InputError, RepositoryClientError
+
+GITHUB_HOSTNAME = "github.com"
+HTTPS_URL_PATTERN = re.compile(rf"^https?:\/\/.*@?{GITHUB_HOSTNAME}\/(.+\/.+?)(.git)?$")
+ACTIONS_USER_NAME = "upload-charms-docs-bot"
+ACTIONS_USER_EMAIL = "upload-charms-docs-bot@users.noreply.github.com"
+ACTIONS_COMMIT_MESSAGE = "migrate docs from server"
+ACTIONS_PULL_REQUEST_TITLE = "[upload-charm-docs] Migrate charm docs"
+ACTIONS_PULL_REQUEST_BODY = (
+    "This pull request was autogenerated by upload-charm-docs to migrate "
+    "existing documentation from server to the git repository."
+)
+PR_LINK_NO_CHANGE = "<not created due to no changes in repository>"
+BRANCH_PREFIX = "upload-charm-docs"
+DEFAULT_BRANCH_NAME = f"{BRANCH_PREFIX}/migrate"
+
+
+class RepositoryClient:
```

Comment:
Should this be in a different file than `pull_request.py` since this could be used for more than creating pull requests?

Reply:
I've thought about having a separate input parser, but it does seem a little far-fetched for this PR to include such refactor(as mentioned in standup).
I think the `DEFAULT_BRANCH_NAME` parameter would fit there nicely, when setting default parameters. However, since I've limited the scope of this PR and currently there's no other reference to this variable other than inside the `pull_request.py` module, i believe it would still belong here.
I'm happy to refactor the code to move this variable somewhere else though. What do you think?